test/hotspot/jtreg/compiler/vectorization/TestEor3AArch64.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2022, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorization;
+
+import jdk.test.lib.Asserts;
+import compiler.lib.ir_framework.*;
+
+import java.util.Random;
+
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8293488
+ * @summary Test EOR3 Neon/SVE2 instruction for aarch64 SHA3 extension
+ * @library /test/lib /
+ * @requires os.arch == "aarch64" & vm.cpu.features ~=".*sha3.*"
+ * @run driver compiler.vectorization.TestEor3AArch64
+ */
+
+public class TestEor3AArch64 {
+
+    private final static int LENGTH = 2048;
+    private final static Random RD = Utils.getRandomInstance();
+
+    private static int[] ia;
+    private static int[] ib;
+    private static int[] ic;
+    private static int[] ir;
+
+    private static long[] la;
+    private static long[] lb;
+    private static long[] lc;
+    private static long[] lr;
+
+    static {
+        ia = new int[LENGTH];
+        ib = new int[LENGTH];
+        ic = new int[LENGTH];
+        ir = new int[LENGTH];
+
+        la = new long[LENGTH];
+        lb = new long[LENGTH];
+        lc = new long[LENGTH];
+        lr = new long[LENGTH];
+
+        for (int i = 0; i < LENGTH; i++) {
+            ia[i] = RD.nextInt(30);
+            ib[i] = RD.nextInt(30);
+            ic[i] = RD.nextInt(30);
+
+            la[i] = RD.nextLong(30);
+            lb[i] = RD.nextLong(30);
+            lc[i] = RD.nextLong(30);
+        }
+    }
+
+    // For the IR verification of "veor3_sve" backend node in the below tests,
+    // "svesha3" CPU feature is supposed to be checked as it is an optional
+    // feature in SVE2 but it is enabled in all the currently available SVE2
+    // hardware. Hence that check is omitted.
+
+    // Test for eor3 Neon and SVE2 instruction for integers
+    @Test
+    @IR(counts = {"veor3_neon", "> 0"}, applyIf = {"MaxVectorSize", "16"})
+    @IR(counts = {"veor3_sve", "> 0"}, applyIfAnd = {"UseSVE", "2", "MaxVectorSize", "> 16"})
```

Comment:
```suggestion
    @IR(counts = {"veor3_sve", "> 0"}, applyIfAnd = {"UseSVE", "2", "MaxVectorSize", "> 16"}, applyIfCPUFeature = {"svesha3", "true"})
```

After this PR(https://github.com/openjdk/jdk/pull/10402), `applyIf` and `applyIfCPUFeature` are evaluated as a logical conjunction. We can check CPU features and VM options at the same time now. Of course, the comment at line 79 should be removed.